pyro/poutine/enumerate_messenger.py

```diff
@@ -32,9 +32,14 @@ def _pyro_sample(self, msg):
             return
 
         if msg["infer"].get("enumerate") == "parallel":
-            # Enumerate over the support of the distribution.
             dist = msg["fn"]
-            value = dist.enumerate_support(expand=msg["infer"].get("expand", EXPAND_DEFAULT))
+            num_samples = msg["infer"].get("num_samples")
+            if num_samples is None:
+                # Enumerate over the support of the distribution.
+                value = dist.enumerate_support(expand=msg["infer"].get("expand", EXPAND_DEFAULT))
+            else:
+                # Monte Carlo sample the distribution.
+                value = dist(sample_shape=(num_samples,))
```

Comment:
Very cool! I think we can just deprecate the global `num_particles` logic now, and use `config_enumerate` instead.

Reply:
I think it's still useful to have a global `num_particles` for some models. For example when there are multiple top-level variables, this PR does not give a way to sample them in a shared tensor dimension, as your parallel particles would:
```py
def model(data):
   x = pyro.sample('x', dist_x())
   y = pyro.sample('y', dist_y())  # does not depend on x
   z = pyro.sample('z', dist_z(x, y))
```

Reply:
Ahh..right. That’s an interesting distinction.